protocol/x/clob/memclob/memclob.go

```diff
@@ -512,6 +512,30 @@ func (m *MemClobPriceTimePriority) PlaceOrder(
 		}
 	}
 
+	// If this is a replacement order, then ensure we remove the existing order from the orderbook.
+	orderId := order.MustGetOrder().OrderId
+	if orderToBeReplaced, found := m.openOrders.getOrder(ctx, orderId); found {
+		m.mustRemoveOrder(ctx, orderToBeReplaced.OrderId)
+		if orderToBeReplaced.IsStatefulOrder() && !m.operationsToPropose.IsOrderRemovalInOperationsQueue(orderToBeReplaced.OrderId) {
+			m.operationsToPropose.MustAddOrderRemovalToOperationsQueue(
+				orderToBeReplaced.OrderId,
+				types.OrderRemoval_REMOVAL_REASON_UNSPECIFIED,
+			)
+		}
```

Comment:
we don't add removal to op queue for replacement 

in the original logic, there is a `makerOrderId != order.MustGetOrder().OrderId` check

Reply:
Can you link that? It seems like we do it. See line 839 on old commit.

Reply:
line 820 in old commit

Reply:
839 is inside the if statement